### ISSUE TITLE
Allow regex in highlight and/or add ability for custom highlighting function

### ISSUE THREAD

**Issue by Symmettry** [face-hh/griddycode#35]
tryna make a typescript highlight but unfortunately I can't highlight types since it's ": any" but I don't have regex to capture that.

**Comment by face-hh** [comment]
I don't see how that would be possible without modifying the GoDot engine's source code, as `highlight_region` is the only method remotely close to RegEx. I would love to be wrong & be corrected that there is a way, so for now I'll leave this issue opened

**Comment by Symmettry** [comment]
https://docs.godotengine.org/en/stable/classes/class_regex.html

for version 4.2 (the one i'm pretty sure you use):

var regex = RegEx.new()
regex.compile("\\w-(\\d+)")

**Comment by Symmettry** [comment]
And also the custom highlighting function would be useful outside of this anyway

**Comment by face-hh** [comment]
Yes, the RegEx resource exists in Godot, however the CodeEdit node can only highlight using keywords and regions - no "from index 3 to index 7, color red"

Which, again, goes to:
> I don't see how that would be possible without modifying the GoDot engine's source code

**Comment by Symmettry** [comment]
Ah i see, that’s unfortunate.

**Comment by Symmettry** [comment]
Are you able to remove old highlights? If so you could add custom highlighting which highlights a region matched by regex in the lua script then remove the old matches.
Although i’m unsure how efficient it is…probably should try to get the code edit thing to allow regex / color by index->index

**Comment by face-hh** [comment]
A region must have a symbol as the first match, not just any character.

**Comment by RealPacket** [comment]
[RegEx]: https://docs.godotengine.org/en/stable/classes/class_regex.html
[CodeEdit]: https://docs.godotengine.org/en/stable/classes/class_codeedit.html
I think you could make a work around.
Given a [RegEx] and a [CodeEdit], bind text/source changing to a function,
which once called, will iterate through every match of the given [RegEx], get the start and end, and highlight region.
Maybe I'll make a PR for it (probably?). You could make an extra behavior parameter that defaults to the old behavior.
